### PR TITLE
Meteor 3

### LINKED ISSUE #395: Using new meteor mongo async api
So with meteor 3.0 moving to the new async api I've noticed that this package is not quite ready for it. 

If you run this with the env variable
WARN_WHEN_USING_OLD_API=1

You end up with errors like this:
```

Calling method users.findOne from old API on server.
W20230803-14:05:48.617(-7)? (STDERR)    This method will be removed, from the server, in version 3.
W20230803-14:05:48.617(-7)? (STDERR)    Trace is below:
W20230803-14:05:48.617(-7)? (STDERR) [object Object]
W20230803-14:05:48.617(-7)? (STDERR)     at warnUsingOldApi (packages/mongo/collection.js:24:12)
W20230803-14:05:48.617(-7)? (STDERR)     at hook.findOne (packages/mongo/collection.js:452:5)
W20230803-14:05:48.618(-7)? (STDERR)     at RedisSubscriptionManager.getDoc (packages/cultofcoders:redis-oplog/lib/redis/RedisSubscriptionManager.js:208:30)
W20230803-14:05:48.618(-7)? (STDERR)     at RedisSubscriptionManager.process (packages/cultofcoders:redis-oplog/lib/redis/RedisSubscriptionManager.js:147:28)
W20230803-14:05:48.618(-7)? (STDERR)     at packages/cultofcoders:redis-oplog/lib/mongo/lib/dispatchers.js:20:42
W20230803-14:05:48.618(-7)? (STDERR)     at Array.forEach (<anonymous>)
W20230803-14:05:48.618(-7)? (STDERR)     at packages/cultofcoders:redis-oplog/lib/mongo/lib/dispatchers.js:14:20
W20230803-14:05:48.618(-7)? (STDERR)     at packages/meteor.js:365:18
W20230803-14:05:48.619(-7)? (STDERR)     at Meteor.EnvironmentVariable.EVp.withValue (packages/meteor.js:1389:31)
W20230803-14:05:48.619(-7)? (STDERR)     at dispatchEvents (packages/cultofcoders:redis-oplog/lib/mongo/lib/dispatchers.js:13:30)
W20230803-14:05:48.619(-7)? (STDERR)     at dispatchUpdate (packages/cultofcoders:redis-oplog/lib/mongo/lib/dispatchers.js:63:5)
W20230803-14:05:48.619(-7)? (STDERR)     at hook.update (packages/cultofcoders:redis-oplog/lib/mongo/Mutator.js:200:13)
W20230803-14:05:48.619(-7)? (STDERR)     at hook.update (packages/cultofcoders:redis-oplog/lib/mongo/extendMongoCollection.js:40:35)
W20230803-14:05:48.619(-7)? (STDERR)     at hook.Mongo.Collection.<computed> [as updateAsync] (packages/mongo/collection.js:1004:46)
```

The root problem looks to be this call:

```
getDoc(collection, subscribers, data) {
        const event = data[RedisPipe.EVENT];
        let doc = data[RedisPipe.DOC];

        if (collection._redisOplog && !collection._redisOplog.protectAgainstRaceConditions) {
            // If there's no protection against race conditions
            // It means we have received the full doc in doc

            return doc;
        }

        const fieldsOfInterest = getFieldsOfInterestFromAll(subscribers);

        if (fieldsOfInterest === true) {
            doc = collection.findOne(doc._id);
        } else {
            doc = collection.findOne(doc._id, { fields: fieldsOfInterest });
        }

        return doc;
    }
```

Are there plans to async-ify this package or is there an alternative solution?

### PR DESCRIPTION
**Pull request by matheusccastroo**
Hi!

This PR goal is to have a version that is compatible with Meteor 3.

- [x] Convert code
- [x] Fix tests

Closes #395

### CODE REVIEW

File: lib/cache/ObservableCollection.js [hunk 1]
```diff
@@ -38,28 +36,6 @@ export default class ObservableCollection {
...
-            const { cursor } = this.collection._redisOplog;
-            if (cursor) {
-                const context = DDP._CurrentPublicationInvocation.get();
-                cursor.call(
-                    context,
-                    cursorDescription.options,
-                    cursorDescription.selector
-                );
-            }
-        }
```
Comment (renanc-br) [inline lib/cache/ObservableCollection.js:62]: why remove this checks?

Comment (matheusccastroo) [reply lib/cache/ObservableCollection.js:62]: They were moved to an `init` function, we just removed from the constructor.

File: .prettierrc [hunk 2]
```diff
@@ -0,0 +1,6 @@
+{
+  "semi": true,
+  "singleQuote": false,
```
Comment (StorytellerCZ) [inline .prettierrc:3]: This should be true to reduce the number of prettier changes and make it easier to review.

**Review by StorytellerCZ — CHANGES_REQUESTED** [review]
Please fix the prettier settings to be more inline with the current setup.

**Comment by StorytellerCZ** [comment]
@matheusccastroo any progress on this? As per request I will release this as rc.2 for people's testing.

**Pull request merged**